### PR TITLE
Sr 478 csv download

### PR DESCRIPTION
**Pull request by Sveouu**
CSV download now available for:

-  Monthly food Group
-  Top Food Source
-  Household Histogram
-  Projected Availability 
- Projected Food Source

Export service removes "_sourceObject" column.

CSV download button disabled when no data for chosen mn is available.

### CODE REVIEW

**Comment by Fuhji** [comment]
Testing with Malawi

- [x] monthly food
- [x] top 20 food (had to use madagascar)
- [ ] household dietary supply - empty csv? Had to click 3 times to get download
- [x] projected availability
- [x] projection food sources

File: src/app/pages/quickMaps/pages/baselineDetails/householdSupply/householdSupply.component.html [hunk 1]
```diff
@@ -25,14 +25,17 @@
     </mat-tab>
 
     <mat-tab label="Download">
-      <app-download [chartDownloadPNG]="chartPNG" [chartDownloadPDF]="chartPDF"></app-download>
+      <app-download [chartDownloadPNG]="chartPNG" [chartDownloadPDF]="chartPDF" [csvObject]="csvDownloadData">
+      </app-download>
     </mat-tab>
 
-    <mat-tab label="About {{title}}">
+    <mat-tab label=" About {{title}}">
```
Comment (Fuhji) [inline src/app/pages/quickMaps/pages/baselineDetails/householdSupply/householdSupply.component.html:32]: I know spaces are ignored usually, but was this vscode formatter adding this pre-space?

File: src/app/pages/quickMaps/pages/biomarkers/biomarkerStatus/biomarkerStatus.component.ts [hunk 2]
```diff
@@ -686,23 +688,38 @@ export class BiomarkerStatusComponent implements AfterViewInit {
       });
   }
 
+  // private initialiseMap(mapElement: HTMLElement): L.Map {
```
Comment (Fuhji) [inline src/app/pages/quickMaps/pages/biomarkers/biomarkerStatus/biomarkerStatus.component.ts:691]: delete if not needed

File: src/app/pages/quickMaps/pages/biomarkers/biomarkerStatus/biomarkerStatus.component.ts [hunk 3]
```diff
@@ -686,23 +688,38 @@ export class BiomarkerStatusComponent implements AfterViewInit {
...
-    }).addTo(map);
-    return map;
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return new LeafletMapHelper()
+      .createMap(mapElement)
+      .setDefaultBaseLayer()
+      .setDefaultControls(() => this.areaBounds)
+      .getMap();
   }
+  // private triggerFitBounds(tabIndex: number): void {
```
Comment (Fuhji) [inline src/app/pages/quickMaps/pages/biomarkers/biomarkerStatus/biomarkerStatus.component.ts:707]: delete commented block if not needed

**Review by Fuhji — COMMENTED** [review]
looks good and all works for me except Household dietary supply. not getting any console errors when clicking, just an empty csv.

**Comment by Sveouu** [comment]
> looks good and all works for me except Household dietary supply. not getting any console errors when clicking, just an empty csv.

Thank you.
Have removed comments and fixed Household Histogram csv download.

**Review by Fuhji — APPROVED** [review]

**Review by jon571 — CHANGES_REQUESTED** [review]
I have a few concerns and a few suggestions :-)

Maybe we could have a screen share to talk through them?

**Review by jon571 — APPROVED** [review]

**Pull request merged**